manage_containers.py

```diff
@@ -0,0 +1,179 @@
+                       'clang-9', 'clang-10', 'clang-11', 'clang-12',
+                       'clang-13', 'clang-14', 'clang-15', 'clang-16', 'clang-17',
+                       'all']
+
+class Container:
+    """
+    Represents a Docker container configured for building the Linux kernel with a specified compiler
+
+
+    Class Attributes:
+        sudo_wrapper (str): wrap commands with sudo (if needed)
+        quiet (bool): enable quiet option in Docker build
+
+    Instance Attributes:
+        gcc (str): Version of GCC compiler
+        clang (str): Version of Clang compiler
+        ubuntu (str): Version of Ubuntu
+        id (str): Docker image ID
+    """
+
+    sudo_wrapper = ''
+    quiet = False
+
+    def __init__(self, gcc_version, clang_version, ubuntu_version):
+        self.gcc = gcc_version
+        self.clang = clang_version
+        self.ubuntu = ubuntu_version
+        self.id = self.check()
+
+    def add(self):
+        """Build the Docker container with the specified compilers"""
+        build_args = ['--build-arg', f'GCC_VERSION={self.gcc}',
+                      '--build-arg', f'CLANG_VERSION={self.clang}',
+                      '--build-arg', f'UBUNTU_VERSION={self.ubuntu}',
+                      '--build-arg', f'UNAME={os.getlogin()}',
+                      '--build-arg', f'UID={os.getuid()}',
+                      '--build-arg', f'GID={os.getgid()}',
+                      '-t', f'kernel-build-container:gcc-{self.gcc}',
+                      '-t', f'kernel-build-container:clang-{self.clang}'
+        ]
+
+        if self.quiet:
+            build_args += ['-q']
+        subprocess.run([self.sudo_wrapper, 'docker', 'build', *build_args, '.'],
+                        text=True, check=True)
+        self.check()
+
+    def rm(self):
+        """Remove the Docker container if it exists and not running"""
+        running = subprocess.run(f'{Container.sudo_wrapper} docker ps | '
+                                 f'grep -E "kernel-build-container:'
+                                 f'(gcc-{self.gcc}|clang-{self.clang})" || true',
+                                 shell=True, text=True, check=True, stdout=subprocess.PIPE).stdout
+        if not running:
+            subprocess.run([self.sudo_wrapper, 'docker', 'rmi', '-f', self.id],
+                            text=True, check=True)
+            self.check()
+            return ''
+        return running
+
+    def check(self):
+        """Check if the Docker container built and parses the image ID"""
+        cmd = subprocess.run([self.sudo_wrapper, 'docker', 'images',
+                              f'kernel-build-container:clang-{self.clang}', '--format', '{{.ID}}'],
+                              stdout=subprocess.PIPE,
+                              text=True, check=True)
+        self.id = cmd.stdout.strip()
+        return self.id
+
+def check_group():
+    """Check if the user is in the Docker group, return 'sudo' if not"""
```

Comment:
How about making this a part of the class constructor?

Initially, you can have `sudo_wrapper = None`.

Then in the `__init__()` you can initialize it to `''` or `'sudo'`.

Reply:
Not sure if it's a great idea because we need to check whether sudo_wrapper is None each time a container instance is created. Anyway, I've made it, so you can check whether this is okay:

https://github.com/a13xp0p0v/kernel-build-containers/pull/22/commits/77a71b90776ce3d9e646d87cf2f97d0d76c14a02